fontkit.d.ts

```diff
@@ -273,7 +570,7 @@ export interface Font {
   // Other properties
   numGlyphs: number /** Number of glyphs in the font */;
   characterSet: number[] /** Array of all of the unicode code points supported by the font */;
-  availableFeatures: any[] /** Array of all OpenType feature tags (or mapped AAT tags) supported by the font */;
+  availableFeatures: TypeFeatures /** OpenType feature tags (or mapped AAT tags) supported by the font */;
```

Comment:
This is actually a `string[]`, based on the following:

https://github.com/Hopding/fontkit/blob/6faf4bf7c675b4b3469abbdf16cf9ab593e1a4e3/src/TTFFont.js#L361-L371

https://github.com/Hopding/fontkit/blob/6faf4bf7c675b4b3469abbdf16cf9ab593e1a4e3/src/layout/LayoutEngine.js#L160-L172

We could type this either as a `availableFeatures: string[]` or, better yet, `availableFeatures: (keyof OpenTypeFeatures | keyof AATFeatures)[]`. This second will provide better type information.

Reply:
Note that if we change the inheritance structure (per my previous comment) we could also set this to `availableFeatures: (keyof FontFeaturesArray)[]`.

Reply:
Since I'm fresh to TS, I'm still learning what it's possible to do. Anyway, I got a bit confused where in your comments you say: `availableFeatures: (keyof FontFeaturesArray)[]`, I assume you mean: `availableFeatures: (keyof TypeFeatures)[]` and I've updated the code accordingly. Hopefully I've done it correctly. :)